spec/function.dd

```diff
@@ -1219,39 +1219,59 @@ $(H2 $(LNAME2 parameters, Function Parameters))
-        $(LI It provides more information to the compiler, enabling more
-        error checking and
-        possibly better code generation.)
+        $(P When talking about function parameters, the $(I parameter) is what the function prototype defines,
+        and the $(I argument) is the value that will $(I bind) to it.
+        For example, using the previous definition of $(D read):
         )
+------
+void main()
+{
+    size_t a = 42;
+    int b;
+    int r = read("Hello World", a, b);
+}
+------
+       $(P In this example, the argument $(D "Hello World") gets bind to $(D input),
+       $(D a) gets bind to $(D count) and $(D b) to $(D errno).
+       )
 
     $(TABLE_2COLS Parameter Storage Classes,
     $(THEAD Storage Class, Description)
-    $(TROW $(I none), parameter becomes a mutable copy of its argument)
-
-    $(TROW $(D in), defined as $(D scope const).  However $(D in) has not yet been properly
-    implemented so it's current implementation is equivalent to $(D const).  It is recommended
-    to avoid using $(D in) until it is properly defined and implemented.  Use $(D scope const)
-    or $(D const) explicitly instead.)
-    $(TROW $(D ref), parameter is passed by reference)
-    $(TROW $(D out), parameter is passed by reference and initialized upon function entry with the default value
-    for its type)
+    $(TROW $(I none), The parameter will be a mutable copy of its argument)
+    $(TROW $(D in), The parameter is an input to the function. Input parameters behaves as if they have
+    the $(D const scope) storage classes. Input parameters can be passed by reference.
+    If they get passed by reference$(COMMA) unlike with $(D ref)$(COMMA) rvalues (such as literals)
+    can be bound to them.
+    Types that would trigger a side effect if passed by value (such as types with postblit$(COMMA)
+    copy constructor$(COMMA) or destructor)$(COMMA) and types which cannot be copied$(COMMA)
+    e.g. if their copy constructor is marked as $(D @disable)$(COMMA) will always be passed by reference.
+    Dynamic arrays$(COMMA) classes$(COMMA) associative arrays$(COMMA) function pointers and delegates
+    will always be passed by value$(COMMA) to allow for covariance.
+    If a parameter is of a type that does not fall in one of those categories$(COMMA)
+    whether or not it is passed by reference is implementation defined$(COMMA) and the backend is free
+    to choose the method that will best fit the ABI of the platform.
+    $(B Note: This require the $(D -preview=in) switch available in v2.094.0 or higher.))
+    $(TROW $(D ref), The parameter is an $(I input/output) parameter$(COMMA) passed by reference.
+    It is expected that the function will both read and write to this parameter.)
+    $(TROW $(D out), The argument must be an lvalue$(COMMA) which will be passed by reference and initialized
+    upon function entry with the default value for its type)
```

Comment:
```suggestion
    upon function entry with the `.init` value of its type)
```

I'm not 100% sure this will be clearer, so maybe you can ignore this suggestion.

Reply:
Maybe using both, e.g. "the default value (`T.init`) of its type" ?